spec/rspec/rails/matchers/have_reported_error_spec.rb

```diff
@@ -0,0 +1,173 @@
+require "rspec/rails/matchers/have_reported_error"
+
+RSpec.describe "have_reported_error matcher" do
+  class TestError < StandardError; end
+  class AnotherTestError < StandardError; end
+
+  it "has an reports_error alias" do
+    expect {Rails.error.report(StandardError.new("test error"))}.to reports_error
+  end
+
+  it "warns that passing value expectation doesn't work" do
+    expect {
+      expect(Rails.error.report(StandardError.new("test error"))).to have_reported_error
+    }.to raise_error(ArgumentError, "this matcher doesn’t work with value expectations")
+  end
+
+  describe "basic functionality" do
+    it "passes when an error is reported" do
+      expect {Rails.error.report(StandardError.new("test error"))}.to have_reported_error
+    end
+
+    it "fails when no error is reported" do
+      expect {
+        expect { "no error" }.to have_reported_error
+      }.to fail_with(/Expected the block to report an error, but none was reported./)
+    end
+
+    it "passes when negated and no error is reported" do
+      expect { "no error" }.not_to have_reported_error
+    end
+  end
+
+  describe "error class matching" do
+    it "passes when correct error class is reported" do
+      expect { Rails.error.report(TestError.new("test error")) }.to have_reported_error(TestError)
+    end
+
+    it "fails when wrong error class is reported" do
+      expect {
+        expect {
+          Rails.error.report(AnotherTestError.new("wrong error"))
+        }.to have_reported_error(TestError)
+      }.to fail_with(/Expected error to be an instance of TestError, but got AnotherTestError/)
+    end
+  end
+
+  describe "error instance matching" do
+    it "passes when error instance matches exactly" do
+      expect {
+        Rails.error.report(TestError.new("exact message"))
+      }.to have_reported_error(TestError.new("exact message"))
+    end
+
+    it "passes when error instance has empty expected message" do
```

Comment:
```suggestion
    it "passes any error of the same class if the expected message is empty" do
```

I would also argue this should fail, if you don't care about the message use the class assertion?